### PR TITLE
Upgrade Qingstor  SDK and support private region

### PR DESCRIPTION
**Pull request by zhijian-pro**

### CODE REVIEW

File: pkg/object/qingstor.go [hunk 1]
```diff
@@ -255,23 +255,27 @@ func newQingStor(endpoint, accessKey, secretKey string) (ObjectStorage, error) {
...
-	zone := hostParts[1]
-
+	var bucketName, zone, host string
+	if !strings.HasSuffix(uri.Host, "qingstor.com") {
+		// support private cloud
+		hostParts := strings.SplitN(uri.Host, ".", 2)
+		bucketName, zone, host = hostParts[0], "", hostParts[1]
+	} else {
+		hostParts := strings.SplitN(uri.Host, ".", 3)
+		bucketName, zone, host = hostParts[0], hostParts[1], hostParts[2]
```
Comment (davies) [inline pkg/object/qingstor.go:265]: Should the zone be part of host? Please add an example here.

Comment (zhijian-pro) [reply pkg/object/qingstor.go:265]: This is old logic and has not changed.

endpoint="http://zhjijian-test.sh1a.qingstor.com"

image

image

The zone is indeed part of the host.

Comment (davies) [reply pkg/object/qingstor.go:265]: I meant the `host` should be `sh1a.qingstor.com`, not `qingstor.com`, right?

Comment (zhijian-pro) [reply pkg/object/qingstor.go:265]: ### host does not include region
image

**Pull request merged**